src/dynamic_loading_c_impl.rs

```diff
@@ -0,0 +1,150 @@
+use crate::error::SysError;
+use crate::debug;
+use core::marker::PhantomData;
+use core::mem::{size_of, zeroed};
+
+#[link(name = "dl-c-impl", kind="static")]
+extern "C" {
+    // we use "usize" to reperesent void*: it's opaque
```

Comment:
Why can't we use this https://doc.rust-lang.org/stable/core/ffi/enum.c_void.html ?

Reply:
It's much like "handle" rather than "pointer". 
For example, if it's defined as a pointer, we assume it can be derefed (*p) but actually it can't.
It's much like a handle in windows API.